### PR TITLE
Show UI conditionally

### PR DESCRIPTION
**Pull request by waximabbax**
For: https://github.com/ProfessionalWiki/PageApprovals/issues/34

https://github.com/ProfessionalWiki/PageApprovals/assets/15813104/5b9f66d3-17db-4649-a276-bdcc7bdb6308

### CODE REVIEW

File: src/Application/UseCases/ApprovalUiQuery/ApprovalUiQuery.php [hunk 1]
```diff
@@ -35,12 +37,18 @@ private function getApprovalState( OutputPage $out, bool $showUi ): ?ApprovalSta
...
 	}
 
 	private function isApprovablePage( OutputPage $out ): bool {
-		return $out->isArticle() // TODO: test
+		$hasAssignedCategories = !empty(
+		array_intersect(
+			$out->getCategories(),
+			$this->approverRepository->getAllAssignedCategories()
+		)
+		); // TODO: test for case sensitive categories
```
Comment (JeroenDeDauw) [inline src/Application/UseCases/ApprovalUiQuery/ApprovalUiQuery.php:49]: Better put this in a dedicated private function

File: src/Application/ApproverRepository.php [hunk 2]
```diff
@@ -16,6 +16,11 @@ public function getApproverCategories( int $userId ): array;
 	 */
 	public function getApproversWithCategories(): array;
 
+	/**
+	 * @return string[]
+	 */
+	public function getAllAssignedCategories(): array;
```
Comment (JeroenDeDauw) [inline src/Application/ApproverRepository.php:22]: `getAllCategoriesWithApprovers` seems a bit better as it avoids `assigned` (a new concept, or at least a new name for one)

**Comment by JeroenDeDauw** [comment]
Seems like right direction

File: src/Adapters/DatabaseApproverRepository.php [hunk 3]
```diff
@@ -75,6 +75,25 @@ public function setApproverCategories( int $userId, array $categoryNames ): void
...
+			[],
+			__METHOD__
+		);
+
+		$allCategories = [];
+		foreach ( $result as $row ) {
+			$allCategories = array_merge( $allCategories, $this->deserializeCategories( (string)$row->categories ) );
+		}
+
+		return array_unique( $allCategories );
```
Comment (JeroenDeDauw) [inline src/Adapters/DatabaseApproverRepository.php:94]: What about the one-liner I posted :(

**Review by JeroenDeDauw — APPROVED** [review]

File: src/Adapters/DatabaseApproverRepository.php (hunk 3 above)
Comment (waximabbax) [reply src/Adapters/DatabaseApproverRepository.php:94]: Sorry! I couldn't get back early. I will create a small PR for refactoring.

**Pull request merged**